.github/workflows/docker.yaml

```diff
@@ -26,34 +39,49 @@ jobs:
               -e DREMIOAI_DREMIO__OAUTH_SUPPORTED=false \
           dremio-mcp-server:${{ github.sha }} \
             dremio-mcp-server tools list
-      - name: Install uv
-        run: |
-           curl -LsSf https://astral.sh/uv/install.sh | sh
-           echo "$HOME/.cargo/bin" >> $GITHUB_PATH
-
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.13'
-
       - name: Start container
         run: |
           docker run \
+            --detach \
             -e DREMIOAI_TOOLS__SERVER_MODE=FOR_DATA_PATTERNS \
             -e DREMIOAI_DREMIO__URI=https://fake \
             -e DREMIOAI_DREMIO__OAUTH_SUPPORTED=false \
             -p 6789:6789 \
-            --name mcp --rm \
+            --name mcp \
+            --rm \
             --network host \
           dremio-mcp-server:${{ github.sha }} \
           dremio-mcp-server run --enable-streaming-http --port 6789 --no-log-to-file \
-            --enable-json-logging &
+            --enable-json-logging
 
       - name: Test container
         run: |
           uv run python tests/stremable_http_cli.py --url http://127.0.0.1:6789/mcp --token fake
+      - id: tag
+        name: Create tag
+        run: |
+          git config --global --add safe.directory /__w/dremio-mcp/dremio-mcp
+
+          head_sha=$(git rev-parse --short HEAD)
+          timestamp=$(date -u +"%Y%m%d%H%M%S")
 
-      - name: Stop container
+          # This tag follows the same convention as https://github.com/dremio/opensearch-k8s-operator/blob/main/.github/workflows/publish.yaml#L17
+          tag="${timestamp}-${head_sha}"
+          
+          echo "tag=$tag" | tee -a "$GITHUB_OUTPUT"
+      - env:
+          ACCESS_TOKEN_FILE: /tmp/access_token.txt
+          GCP_METADATA_URL: http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token
+        name: Login to pkg.dev
         run: |
-          docker stop mcp
+          curl -H "Metadata-Flavor: Google" "${GCP_METADATA_URL}" | jq -r '.access_token' > ${ACCESS_TOKEN_FILE}
+          cat ${ACCESS_TOKEN_FILE} | docker login -u oauth2accesstoken --password-stdin us-west1-docker.pkg.dev
+      - name: Push container
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          tags: |
+            us-west1-docker.pkg.dev/dremio-1093/dremio/dremio-mcp-server:${{ steps.tag.outputs.tag }}
```

Comment:
@eduardo-munoz-dremio instead of adding the GAR id here, can we actually make it user triggered (as opposed to automated workflow). The workflow can be triggered by our build automation with the [tags as input](https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows#providing-inputs), or manually if we need it.